src/app/components/challenge/challengeleaderboard/challengeleaderboard.component.ts

```diff
@@ -411,4 +428,35 @@ export class ChallengeleaderboardComponent implements OnInit, AfterViewInit {
       () => {}
     );
   }
+
+  showLeaderboardByLatestOrBest() {
+    const API_PATH = this.endpointsService.particularChallengePhaseSplitUrl(this.selectedPhaseSplit['id']);
+    const SELF = this;
+    let BODY;
+
+    if (this.phaseSplits[this.selectedPhaseSplit['id']].hasOwnProperty('show_leaderboard_by_latest_submission')) {
+      BODY = JSON.stringify({
+        'show_leaderboard_by_latest_submission': this.phaseSplits[this.selectedPhaseSplit['id']].show_leaderboard_by_latest_submission
+      });
+    } else {
+      BODY = JSON.stringify({
+        'show_leaderboard_by_latest_submission': SELF.showLeaderboardByLatest
+      });
+    }
+
+    SELF.apiService.patchUrl(
+      API_PATH,
+      BODY
+    ).subscribe(
+      data => {
+        SELF.sortLeaderboardTextOption = SELF.showLeaderboardByLatest ? 'Sort by best' : 'Sort by latest';
+        SELF.showLeaderboardByLatest = !SELF.showLeaderboardByLatest;
```

Comment:
You should move this above line 452 cause based on this variable only you are updating `SELF.sortLeaderboardTextOption`.

Reply:
changed!